iot-e2e-tests/common/src/test/java/tests/integration/com/microsoft/azure/sdk/iot/iothub/setup/DirectMethodsCommon.java

```diff
@@ -221,13 +275,88 @@ public void openDeviceClientAndSubscribeToMethods() throws Exception
                     result = e.toString();
                     status = METHOD_THROWS;
                 }
-                deviceMethodData = new DirectMethodResponse(status, result);
+                deviceMethodData = new com.microsoft.azure.sdk.iot.device.twin.DirectMethodResponse(status, result);
 
                 return deviceMethodData;
             },
             null);
     }
 
+    public void subscribeToMethodAndReceiveAsDifferentTypes(String type) throws Exception
```

Comment:
Reconstruct this method to take payload type as parameter and create DirectMethodResponse object accordingly.